src/session/mod.rs

```diff
@@ -500,7 +500,24 @@ impl Session {
 
     pub(crate) fn select_chat(&self, chat_id: i64) {
         let imp = self.imp();
-        imp.sidebar.set_selected_chat(Some(self.chat(chat_id)));
+        let client_id = self.client_id();
+
+        if let Some(chat) = self.try_chat(chat_id) {
+            imp.sidebar.select_chat(chat);
+        } else {
+            spawn(clone!(@weak self as obj => async move {
+                let imp = obj.imp();
+
+                match functions::create_private_chat(chat_id, true, client_id).await {
+                    Ok(enums::Chat::Chat(data)) => {
+                        imp.sidebar.select_chat(obj.chat(data.id));
+                    }
+                    Err(e) => {
+                        log::warn!("Failed to create private chat: {:?}", e);
+                    }
+                }
+            }));
+        }
```

Comment:
The code could be a bit cleaner by using `match` and keep variables as local as possible
```suggestion
        match self.try_chat(chat_id) {
            Some(chat) => imp.sidebar.select_chat(chat),
            None => spawn(clone!(@weak self as obj => async move {
                match functions::create_private_chat(chat_id, true, obj.client_id()).await {
                    Ok(enums::Chat::Chat(data)) => obj.imp().sidebar.select_chat(obj.chat(data.id)),
                    Err(e) => log::warn!("Failed to create private chat: {:?}", e),
                }
            })),
        }
```